renpy/common/_developer/developer.rpym

```diff
@@ -208,6 +210,66 @@ label _debugger_screen:
 
     return
 
+screen _persistent_data_viewer(entries):
+
+    zorder 1010
+    modal True
+
+    frame:
+        style_prefix ""
+        right_padding 0
+
+        has side "t c b":
+            spacing gui._scale(10)
+
+        label _("Persistent Data Viewer")
+
+        viewport:
+            scrollbars "both"
+            child_size (4000, 0)
+            yfill True
+            xfill True
+            mousewheel True
+
+            vbox:
+
+                if not entries:
+                    text _("Nothing to inspect.")
+
+                for vn, value in entries:
+                    text "[vn] = [value!q]"
+
+        hbox:
+            spacing gui._scale(25)
+
+            textbutton _(u"Return"):
+                action Return(True)
+
+label _persistent_explorer:
+
+    call _enter_game_menu
+
+    python hide:
+        import reprlib
+        aRepr = reprlib.Repr()
+        aRepr.maxstring = 120
+
+        entries = []
+
+        for vn in sorted(persistent._keys()):
```

Comment:
```suggestion
        for vn in sorted(persistent.__dict__):
```

This would alleviate the need to alter `renpy/persistent.py`, and allow the display of variables with a leading double underscore as line 263 seeks to do.

Reply:
The screen and label are mainly copypaste from the variable viewer that I edited the least possible to make replacing them easier. But I don't intend any underscored things to be visible. I'll probably comment these lines.
Unless you have a reason to make double-underscored elements visible, but since the persistent inner process doesn't treat them normally (see former line 69) I don't think we should support it.

Reply:
While probably unintentional, it's possible use a leading double underscore to set an (as per line 69) persistent value that won't be cleared unless done so explicitly, there are reasonable uses for that, especially for games that provide the ability for users to reset their persistent data ("reset statistics" type of thing) where it makes sense to clear practically but perhaps not quite all data.

It's certainly not a "take it under our wing and say it's supported" thing, so much as a "it currently works and there's no real need to break it" thing, and with that being the case showing such values in the inspector wouldn't seem to do any harm while being potentially useful for those making use of said quirk. That it also aligns with the existing logic of the current variable viewer is a happy bonus.

Reply:
I disagree about showcasing undocumented things in a devtool. If you want to document the __ thing (with the name-mangling mechanism on top of it as always), it should be done separately.

Reply:
IMO hiding values that could exist for purely idealogical reasons seems to run counter to it being a tool to help creators.

Reply:
As a developer, I'd like to be able to see everything via the dev tool so I can debug every possible case. I think it might be beneficent even for renpy itself.

Reply:
Some persistent values are used internally and should not be visible for creators, for the same kind of reasons the `config`, `preferences` and so on stores are not visible in the normal variable viewer.
If you want to start considering __-started persistent variables as a documented features for creators, then create a PR proposing it, documenting it and adressing how it works with the name-mangling mechanism (which is all but simple to understand).
Until then, we should not advertize or facilitate the use of undocumented features.

Reply:
I think you're conflating two things here. Just because a something isn't documented isn't a reason to hide it, it's a reason to stay neutral/unopinionated about it. Actively hiding things because support for them isn't documented isn't helpful to anyone, and having it be displayed isn't any form of documentation or binding agreement to support it, especially in a devtool. It's just allowing people to see what values they put there, even if nothing said they could (because it's not documented).

Reply:
I understand your argument, but I also disagree with it. I sort-of agree the __-starting keys should be adressed one way or another, but it won't be in this PR.